plugins/modules/ssh_config.py

```diff
@@ -214,19 +246,17 @@ def ensure_state(self):
             port=self.params.get('port'),
             identity_file=self.params.get('identity_file'),
             user=self.params.get('remote_user'),
-            strict_host_key_checking=self.params.get('strict_host_key_checking'),
+            strict_host_key_checking=fix_bool_str(self.params.get('strict_host_key_checking')),
```

Comment:
Why is this needed? `strict_host_key_checking` can only take the three string values `'yes'`, `'no'`, and `'ask'`. It can never be `'True'` or `'False'`.

Reply:
When setting this value in the YAML playbook to `yes` instead of `"yes"`, YAML interprets this as boolean, Python will use `bool` (`True`) type, the module class converts it to `str` (`"True"`) as per `type` specification. While this is a user error of using the wrong data type, it is very common and difficult to spot.

Reply:
`strict_host_key_checking` has `choices` set (https://github.com/ansible-collections/community.general/blob/4df626dd5d3e293e9219f05d1aaafd428ee9a025/plugins/modules/ssh_config.py#L355), so `AnsibleModule` will make sure that its value is exactly one of these strings.

The argspec validator actually has special code to handle converting `'False'` and `'True'` back to 'equivalent' choices strings (in this case: `yes` and `no`): https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/common/parameters.py#L672-L683 If it wouldn't have this code, the module would fail if you provide a boolean here (since `'True'` and `'False'` are not on the choices list).

Reply:
The second commit should not be dropped without a replacement since `controlpersist` can use such a conversion. Since it has `type='str'` without `choices` you end up with `'True'` and `'False'` when passing booleans.